### PR TITLE
Make markers immutable

### PR DESCRIPTION
**Pull request by dajudge**
We figured for a couple of reasons the log format marker should be immutable:
* it seems to be what people expect
* re-using a base-marker to create sub-markers is not possible when the same mutable object is re-used
* there's probably more

### CODE REVIEW

File: src/test/java/com/atomicjar/logging/LogFmtMarkerTest.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.atomicjar.logging;
+
+import org.junit.jupiter.api.Test;
+
+public class LogFmtMarkerTest {
+    @Test
+    public void testImmutability() {
+        var orig = LogFmtMarker.with("a", "b");
+        orig.and("c", "d");
+        assert orig.getData().size() == 1;
```
Comment (eddumelendez) [inline src/test/java/com/atomicjar/logging/LogFmtMarkerTest.java:10]: using assertj

```suggestion
        assertThat(orig.getData()).hasSize(1);
```

Comment (dajudge) [reply src/test/java/com/atomicjar/logging/LogFmtMarkerTest.java]: done

File: src/test/java/com/atomicjar/logging/LogFmtMarkerTest.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.atomicjar.logging;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class LogFmtMarkerTest {
```
Comment (eddumelendez) [inline src/test/java/com/atomicjar/logging/LogFmtMarkerTest.java:7]: junit 5 test classes don't need to be public

```suggestion
class LogFmtMarkerTest {
```

**Review by eddumelendez — APPROVED** [review]
LGTM! 👍🏽

**Pull request merged**